index.html

```diff
@@ -16,162 +16,238 @@
-                displayImage.src = dt;
-
-            };
-
-            function dlCanvas()
-            {
-                var dt = displayImage.src;
-                this.href = dt;
-            };
-
-            function copyClipboardCanvas()
-            {
-                var dt = previewCanvas.toDataURL('image/png');
-
-                displayImage.src = dt;
-            };
-
-            window.onload = function()
-            {
-                dl.addEventListener('click', dlCanvas, false);
-
-                var form = document.getElementById("controlForm");
-                form.onsubmit = function() {return false};
-                form.onchange = refreshImage;
-
-                refreshImage();
-            };
-        </script>
-    </head>
-    <body>
-	    <div class="mainBody">
-			<h1>Perlin Noise Maker</h1>
-			<p>Create your own images of Perlin noise!  This app will generate tileable Perlin noise textures which is a useful raw material for may image processing applications.</p>
-
-			<p>Adjust the values below to change the proerties of the image.</p>
-			<p>To save the image, click on the <i>Download Image</i> link below.  Alternately, you can right click the image and use your web browser's menu to save it to disk.</p>
-
-			<img id="displayImage" style="border:1px solid #000000;"
-				 alt="noise.png"/>
-
-			<!--
-			<canvas id="previewCanvas" width="256" height="256" style="border:1px solid #000000;"></canvas>
-			-->
-
-			<script>
-			</script>
-
+<head>
+	<title>Noise Maker</title>
+	<meta charset="UTF-8">
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
+	<link rel="stylesheet" type="text/css" href="noiseMaker.css">
+	<script src="noiseMaker.js"></script>
+	<script>
+		function refreshImage()
+		{
+			var spec = {};
+
+			spec.randseed = document.getElementById("randseed").value;
+			spec.period = document.getElementById("period").value;
+			spec.levels = document.getElementById("numLevels").value;
+			spec.atten = document.getElementById("atten").value;
+			spec.absolute = document.getElementById("absolute").checked;
+			spec.color = document.getElementById("noiseColor").checked;
+			spec.alpha = document.getElementById("noiseAlpha").checked;
+
+			var imgWidth = document.getElementById("imgWidth").value;
+			var imgHeight = document.getElementById("imgHeight").value;
+
+			var c = document.createElement('canvas');
+			c.width = imgWidth;
+			c.height = imgHeight;
+
+			var ctx = c.getContext("2d");
+
+			var img = new Image(imgWidth, imgHeight);
+			//img.createChecker(period);
+			//                img.createPerlinNoise(period, randseed);
+			img.createTurbulence(spec);
+			var imgData = img.toImageContext(ctx);
+			ctx.putImageData(imgData, 0, 0);
+
+			var dt = c.toDataURL('image/png');
+			displayImage.src = dt;
+
+		};
+
+		function dlCanvas()
+		{
+			var dt = displayImage.src;
+			this.href = dt;
+		};
+
+		// Requires save connection (HTTPS)
+		function copyClipboardCanvas()
```

Comment:
Potential implementation of copying to clipboard implemented, but in order for browsers to actually copy those files into the clipboard they require a HTTPS connection, there for a implementation that doesn't work at the moment, because the website is missing a security certificate. (disabled by setting the buttons style to "display:none;" in line 196)